### PR TITLE
[Fix] Fixed bug #31

### LINKED ISSUE #31: [BUG] Browser.profiles() returns only the directory name
**Describe the bug**

In `generic.Browser`, the `profiles()` function returns only the last part of the path (the directory name).
For example, if a browser has profiles stored as such: `profiles/profile1`, `profiles/profile2`, etc. `profiles()` will return `["profile1", "profile2"]` which will fail in `fetch()` since the intermediate directory is missing.

This does not affect any of the existing browsers, but it's a bug that needs to be addressed before such a browser is added.

**To Reproduce**

Run the following code from the directory of this repo. 

```python
from browser_history.generic import Browser
from pathlib import Path

# to ensure that it looks in the current dir instead of home dir
Path.home = lambda: Path(".")

class SomeBrowser(Browser):
    name = "Non-existent browser"

    windows_path = "./tests/test_homedirs/Linux/.mozilla/"
    mac_path = "./tests/test_homedirs/Linux/.mozilla/"
    linux_path = "./tests/test_homedirs/Linux/.mozilla/"

    profile_support = True
    profile_dir_prefixes = ["*/*"]
    history_file = "places.sqlite"

s = SomeBrowser()
print(s.profiles())
print(s.history_paths())
print(s.fetch()) # Error
```

It fails with an error since the intermediate directory `firefox` was not returned by `profiles()`

**Expected behavior**

`profiles()` should return the entire *relative* path from `history_dir` instead of just the last directory name.

**Additional context**

Related code:

https://github.com/pesos/browser-history/blob/b371317d51812068c7e6c32b2a16fead62aba185/browser_history/generic.py#L85-L101

### PR DESCRIPTION
**Pull request by lucasace**
# Description
I fixed the directory issue, removed maybe_profile_dirs, as it wasnt that important with this change, 
File can be found any where deep in a directory given the history_dir and history_file name

Fixes #31 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] I have read the [contribution guidelines](https://browser-history.readthedocs.io/en/latest/contributing.html) and followed it as far as possible. 
- [x] I have performed a self-review of my own code (if applicable)
- [ ] I have commented my code, particularly in hard-to-understand areas (if applicable)
- [ ] I have made corresponding changes to the documentation (if applicable)
- [x] My changes generate no new warnings 
- [x] Any dependent and pending changes have been merged and published

### CODE REVIEW

**Review by Samyak2 — CHANGES_REQUESTED** [review]
Looks good. Just a few things.
Some parts of the code do not look good, the formatting I mean. I would suggest running [black](https://github.com/psf/black) on it. (I will make this a requirement for PRs sometime soon)

File: browser_history/generic.py [hunk 1]
```diff
@@ -84,20 +84,15 @@ def __init__(self, plat: utils.Platform = None):
...
-        for profile_dir_prefix in self.profile_dir_prefixes:
-            maybe_profile_dirs.extend(self.history_dir.glob(profile_dir_prefix))
-        profile_dirs = [profile_dir.name for profile_dir in maybe_profile_dirs
-                        if (profile_dir / self.history_file).exists()]
-
+        file =glob.glob(str(self.history_dir)+"/**/"+str(self.history_file),recursive=True)
+        profile_dirs=[]
+        for profile_dir in file:
+            profile_dirs.append(profile_dir.split(str(self.history_dir)+'/')[-1]\
+                .split('/'+self.history_file)[0])
```
[the remark this answers is not in the document]
Comment (lucasace) [reply browser_history/generic.py:95]: Okay sure, Will do

[the remark this answers is not in the document]
Comment (lucasace) [reply browser_history/generic.py]: I completed the changes

File: browser_history/generic.py [hunk 2]
```diff
@@ -84,20 +86,20 @@ def __init__(self, plat: utils.Platform = None):
...
-
+            return ["."]
+        file = glob.glob(
+            str(self.history_dir) + "/**/" + str(self.history_file), recursive=True
+        )
+        profile_dirs = []
+        for profile_dir in file:
+            profile_dirs.append(
+                profile_dir.split(str(self.history_dir) + "/", maxsplit=2)[-1].split(
+                    "/" + self.history_file, maxsplit=2
```
Comment (Samyak2) [inline browser_history/generic.py:100]: Here the `maxsplit` should be 1, right? Since we need exactly 2 items to be returned.

**Review by Samyak2 — CHANGES_REQUESTED** [review]
(Sorry for the late response)

File: browser_history/generic.py (hunk 2 above)
Comment (lucasace) [reply browser_history/generic.py:101]: ohh yeahh, I got confused  , I'll change it right away

Comment (lucasace) [reply browser_history/generic.py]: done

**Pull request merged**